$OpenBSD: patch-src_merge-evolution_c,v 1.1 2012/10/26 06:49:05 jasper Exp $

Fix glabels broken with evolution-data-server 3.6.0
https://bugzilla.gnome.org/show_bug.cgi?id=685130

Patch from Fedora:
http://pkgs.fedoraproject.org/cgit/glabels.git/tree/glabels-3.0.0-new-eds.patch

--- src/merge-evolution.c.orig	Tue Sep 18 02:01:02 2012
+++ src/merge-evolution.c	Fri Oct 26 08:43:05 2012
@@ -29,7 +29,7 @@
 
 #include "merge-evolution.h"
 
-#include <libebook/e-book.h>
+#include <libebook/libebook.h>
 #include <glib/gi18n.h>
 #include <stdio.h>
 #include <string.h>
@@ -297,6 +297,31 @@ static gint sort_contact_by_file_as(gconstpointer *a, 
 /*--------------------------------------------------------------------------*/
 /* Open merge source.                                                       */
 /*--------------------------------------------------------------------------*/
+static EBook *
+gl_open_system_addressbook (GError **error)
+{
+   ESourceRegistry *registry;
+   EBook *book = NULL;
+   ESource *source;
+
+   registry = e_source_registry_new_sync (NULL, error);
+   if (!registry)
+       return NULL;
+
+   source = e_source_registry_ref_builtin_address_book (registry);
+   if (!source) {
+       g_object_unref (registry);
+       return NULL;
+   }
+
+   book = e_book_new (source, error);
+
+   g_object_unref (source);
+   g_object_unref (registry);
+
+   return book;
+}
+
 static void
 gl_merge_evolution_open (glMerge *merge)
 {
@@ -317,12 +342,12 @@ gl_merge_evolution_open (glMerge *merge)
                 return;
         }
 
-        merge_evolution->priv->book = e_book_new_system_addressbook(&error);
+        merge_evolution->priv->book = gl_open_system_addressbook(&error);
         if (!merge_evolution->priv->book) {
                 g_warning ("Couldn't open addressbook.");
                 if (error)
                 {
-                        g_warning ("e_book_new_system_addressbook: %s", error->message);
+                        g_warning ("gl_open_system_addressbook: %s", error->message);
                         g_error_free (error);
                 }
                 e_book_query_unref(query);
